doc/source/math/resizing_ops.rst

```diff
@@ -0,0 +1,319 @@
+On the other hand, :math:`R` has no left inverse since it has a non-trivial kernel (null space) :math:`\mathrm{ker} R = \{x \in \mathbb{R}^n\,|\,x_i = 0 \text{ for } i = p, \dots, p+m-1\}`.
+
+
+Extension operators
+===================
+Now we study the opposite case of resizing, namely extending a vector.
+We thus choose :math:`m > n` and consider different cases of enlarging a given vector :math:`x \in \mathbb{R}^n` to a vector in :math:`\mathbb{R}^m`.
+The start index is again denoted by :math:`p` and needs to fulfill :math:`0 \leq p \leq m - n - 1`, such that a vector of length :math:`n` "fits into" a vector of length :math:`m` when starting at index :math:`p`.
+
+It should be noted that all extension operators mentioned here are of the above form :math:`R^* + P` with :math:`P` acting on the "outer" indices only.
+Hence they all act as a right inverses for the restriction operator.
+This property can also be read as the fact that all extension operators are left-inverted by the restriction operator :math:`R`.
+
+Moreover, the "mixed" case, i.e. the combination of restriction and extension which would occur e.g. for a constant index shift :math:`x \mapsto (0, \dots, 0, x_0, \dots, x_{n-p-1})`, is not considered here.
+It can be represented by a combination of the two "pure" operations.
+
+
+Zero padding
+------------
+In this most basic padding variant, one fills the missing values in the target vector with zeros, yielding the operator
+
+.. math::
+    E_{\mathrm{z}} : \mathbb{R}^n \to \mathbb{R}^m, \quad E_{\mathrm{z}}(x)_j :=
+    \begin{cases}
+    x_{j-p}, & \text{if } p \leq j \leq p + n - 1, \\
+    0      , & \text{else}.
+    \end{cases}
+    :label: def_zero_pad_op
+
+Note that this is the adjoint of the restriction operator :math:`R` defined in :eq:`def_restrict_op`.
+Hence, its adjoint is given by the restriction, :math:`E_{\mathrm{z}}^* = R`.
+
+
+Constant padding
+----------------
+In constant padding with constant :math:`c`, the extra zeros in :eq:`def_zero_pad_op` are replaced with :math:`c`.
+Hence, the operator performing constant padding can be written as :math:`E_{\mathrm{c}} = E_{\mathrm{z}} + P_{\mathrm{c}}`, where the second summand is given by
+
+.. math::
+    P_{\mathrm{c}}(x) =
+    \begin{cases}
+    0      , & \text{if } p \leq j \leq p + n - 1, \\
+    c      , & \text{else}.
+    \end{cases}
+
+Note that this operator is not linear, and its derivative is the zero operator, hence the derivative of the constant padding operator is :math:`E_{\mathrm{c}}' = E_{\mathrm{z}}`.
+
+
+Periodic padding
+----------------
+This padding mode continues the original vector :math:`x` periodically in both directions.
+For reasons of practicability, at most one whole copy is allowed on both sides, which means that the numbers :math:`n`, :math:`m` and :math:`p` need to fulfill :math:`p \leq n` ("left" padding amount) and :math:`m - (p + n) \leq n` ("right" padding amount).
```

Comment:
Relaxing this is certainly possible, but it would be a significant extra complexity. Instead of one slice left and right per axis, one would have to use lists, which I don't really feel like thinking about now. This is already complex enough, and if we really need multiple periods at some point, we can add it. Already now, you can get a poor man's version of that by simply extending in several steps.